### PR TITLE
T2D-267 google OAuth for properties Registry

### PR DESCRIPTION
**Pull request by selvaebi**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/ampt2d/registry/entities/User.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Id;
+
+@Entity
+public class User {
+
+    public enum Roles {
```
Comment (Zhicheng-Liu) [inline src/main/java/uk/ac/ebi/ampt2d/registry/entities/User.java]: `Role`, singular form for class names.

File: src/test/java/uk/ac/ebi/ampt2d/registry/PropertyRegistryServiceApplicationTests.java [hunk 2]
```diff
@@ -236,11 +246,45 @@ public void testPaging() throws Exception {
...
     }
 
+    @Test
+    public void testAuthorization() throws Exception {
+        // Any url other than root and swagger is Secured
+        mockMvc.perform(get("/phenotypes")).andExpect(status().isUnauthorized());
+        mockMvc.perform(get("/properties")).andExpect(status().isUnauthorized());
+
+        mockMvc.perform(get("/")).andExpect(status().isOk()); // Root is not secured
+        mockMvc.perform(get("/swagger-ui.html")).andExpect(status().isOk()); // Swagger is not secured
```
Comment (Zhicheng-Liu) [inline src/test/java/uk/ac/ebi/ampt2d/registry/PropertyRegistryServiceApplicationTests.java]: For completeness, could you test all the URLs in the whitelist?

File: src/test/java/uk/ac/ebi/ampt2d/registry/PropertyRegistryServiceApplicationTests.java [hunk 3]
```diff
@@ -236,11 +246,45 @@ public void testPaging() throws Exception {
...
+        String content = "{\"id\":\"CALL_RATE\"," +
+                "\"type\":\"FLOAT\"," +
+                "\"meaning\":\"CALL_RATE\"," +
+                "\"description\":\"calling rate\"}";
+
+        //POST can be performed  by EDITOR or ADMIN only
+        mockMvc.perform(post("/properties").with(oAuthHelper.bearerToken("testUser@gmail.com"))
+                .content(content)).andExpect(status().isForbidden());
+        mockMvc.perform(post("/properties").with(oAuthHelper.bearerToken("testEditor@gmail.com"))
+                .content(content)).andExpect(status().isCreated());
```
Comment (Zhicheng-Liu) [inline src/test/java/uk/ac/ebi/ampt2d/registry/PropertyRegistryServiceApplicationTests.java]: For completeness, I would also add test cases for other request methods and roles combinations.

**Review by Zhicheng-Liu — CHANGES_REQUESTED** [review]

**Review by Zhicheng-Liu — APPROVED** [review]

File: src/test/resources/application.properties [hunk 4]
```diff
@@ -9,4 +9,6 @@ management.security.enabled=false
 management.context-path=/actuator
 management.endpoint.health.show-details=always
 endpoints.health.sensitive=false
-registry.protocols=http
\ No newline at end of file
+registry.protocols=http
```
Comment (iyangar) [inline src/test/resources/application.properties:12]: Do you mean `https`?

**Review by iyangar — CHANGES_REQUESTED** [review]

File: src/test/resources/application.properties (hunk 4 above)
Comment (selvaebi) [reply src/test/resources/application.properties:12]: no test uses http

Comment (iyangar) [reply src/test/resources/application.properties:12]: ok, fine. same line was deleted and added again, hence got clarified

**Review by iyangar — APPROVED** [review]

**Pull request merged**